### PR TITLE
Added impersonate header

### PR DESCRIPTION
**Pull request by Skeen**
This PR introduces impersonation via a HTTP Header as suggested on #119.
This allows admins to impersonate other users, fetching their data, etc.

Currently this is a backend only implementation to support integrations.

### CODE REVIEW

File: auth/middleware.go [hunk 1]
```diff
@@ -51,33 +52,55 @@ func sessionCallbacks(request *http.Request, writer http.ResponseWriter) *http.R
 }
 
 func reqisterUser(request *http.Request, writer http.ResponseWriter, db *gorm.DB) *http.Request {
```
Comment (jmattheis) [inline auth/middleware.go:54]: Hey, the code looks fine, could you add some tests in auth/middleware_test.go?

Comment (Skeen) [reply auth/middleware.go:54]: I added some HTTP status-code returns and a few tests.

**Review by jmattheis — APPROVED** [review]
Looks good, I fixed the formatting via `go fmt`.

File: auth/middleware.go [hunk 2]
```diff
@@ -51,33 +52,66 @@ func sessionCallbacks(request *http.Request, writer http.ResponseWriter) *http.R
...
-	if token, err := getToken(request); err == nil {
-		device := &model.Device{}
-		if db.Where("token = ?", token).Find(device).RecordNotFound() {
-			log.Info().Str("token", token).Msg("no device with token found")
+	token, err := getToken(request)
+	if err != nil {
+		msg := "Issue fetching token"
+		log.Info().Msg(msg)
+		http.Error(writer, msg, 400)
```
Comment (Skeen) [inline auth/middleware.go:59]: I think this is problematic @jmattheis.

I'm seeing output like this (on http://localhost:3030/):
```
Issue fetching token
<!doctype html><html lang="en"><head><meta charset="utf-8"/>...</head><body>...</body></html>
```
Do you see the same issue on your end?

Comment (jmattheis) [reply auth/middleware.go:59]: Yeah, fixed in the last commit.

**Pull request merged**